### DISCUSSION TITLE
Problem with Drawing Feature in ngx-extended-pdf-viewer in Angular 16

### DISCUSSION THREAD

**Discussion by 94ostry** [stephanrauh/ngx-extended-pdf-viewer#2608] [General]
Hello everyone,

I'm using the `ngx-extended-pdf-viewer:21.4.6` component in my Angular 16 application, and I've encountered an issue with the drawing feature. When I start drawing on a PDF, it suddenly stops working, and I can’t continue drawing. I have to toggle the Draw button on and off to make it work again, and sometimes a physical restart of the application is required.

I had filling that it is sth related with Angular 16 😢 

Has anyone else experienced a similar problem or knows how to resolve this? Any help would be appreciated!

Thanks in advance!

**Comment by schtucks** [comment]
I have the exact same behaviour with the same versions as him ....

Any solution yet?

**Comment by wider2** [comment]
The same problem in angular 16
"ngx-extended-pdf-viewer": "^21.4.6",

if i try to draw a line or highlight the text - i can do it only once . After that this error occurs

```
Uncaught TypeError: Cannot read properties of null (reading 'isEmpty')
    at #Ma (viewer-4.6.688.min.mjs:23:296510)
```

here in code:
```js
            this._freeHighlight.isEmpty() ? t.drawLayer.removeFreeHighlight(this._freeHighlightId) : t.createAndAddNewEditor(e, !1, {
                highlightId: this._freeHighlightId,
                highlightOutlines: this._freeHighlight.getOutlines(),
                clipPathId: this._freeHighlightClipId,
                methodOfCreation: "main_toolbar"
```

**Reply by stephanrauh** [reply]
I've reluctantly dropped support for Angular 16 because too many people reported problems. However, this doesn't sound like an Angular problem. Are you sure it goes away by updating to Angular 17?

Be that as it may, please set the attribute `[minifiedJSLibraries]="false"` and send me the stack trace again. That simplifies things tremendously.

**Reply by stephanrauh** [reply]
I suspect the error occurs here?

https://github.com/stephanrauh/pdf.js/blob/b2f48e0961143a707a68138e4d5cf76cabeb0ce8/src/display/editor/highlight.js#L799-L813

**Reply by 94ostry** [reply]
I tested with Angular 17 and works 🎉

**Comment by wider2** [comment]
yes it was that method highlight

```
TypeError: Cannot read properties of null (reading 'isEmpty')
  at #endHighlight (/assets/viewer-4.6.688.mjs:18036:30)
  at pointerUpCallback (/assets/viewer-4.6.688.mjs:18002:25)
  at _ZoneDelegate.invokeTask (/polyfills.js:10773:173)
  at Zone.runTask (/polyfills.js:10563:39)
  at ZoneTask.invokeTask [as invoke] (/polyfills.js:10851:28)
```

now if i added property to the library declaration `<ngx-extended-pdf-viewer
 [minifiedJSLibraries]="false"`

`isEmpty()` not failed anymore in developer console

but as before, i can draw a line only once 
and highlight only once

it seems to be an array can store only latest value?

**Reply by stephanrauh** [reply]
Maybe downgrading to version 21.4.4 or upgrading to version 22 solves the issue. I've messed up the later 21.4.* versions. The automated merge conflict resolution was bad, and I didn't notice for several months.

**Reply by 94ostry** [reply]
Test with version 22 and Angular 17. The problem does not occur
